pkg/cache/clusterqueue.go

```diff
@@ -488,7 +488,7 @@ func (c *clusterQueue) updateWorkloadUsage(wi *workload.Info, m int64) {
 			updateFlavorUsage(frUsage, lq.admittedUsage, m)
 			lq.admittedWorkloads += int(m)
 		}
-		if features.Enabled(features.LocalQueueMetrics) {
+		if features.Enabled(features.LocalQueueMetrics) && metrics.ShouldReportLocalMetrics(lq.labels) {
```

Comment:
you already check the feature gate in the function. Please update everywhere
```suggestion
		metrics.ShouldReportLocalMetrics(lq.labels) {
```